File: bm_db_android_kotlin/app/src/main/java/steffen/de/bm_db_android_kotlin/ViewModel/MainActivityViewModel.kt
package steffen.de.bm_db_android_kotlin.ViewModel

import android.app.Application
import kotlinx.coroutines.experimental.launch
import kotlinx.coroutines.experimental.runBlocking
import steffen.de.bm_db_android_kotlin.Model.Data
import steffen.de.bm_db_android_kotlin.Database.DataDatabase
import kotlin.system.measureTimeMillis

class MainActivityViewModel(application: Application, private var database: DataDatabase = DataDatabase.getDatabase(application)){

    private var dataList: MutableList<Data>? = mutableListOf<Data>()

    init {
        addDataToList()
    }

    fun loadData() = runBlocking{
        val startTime = System.nanoTime();

        val job = launch{
            dataList = database.dataDao().all
        }
        job.join()

        val endTime = System.nanoTime();
        println("Time: Load -> " + ((endTime - startTime) / 1000000))
        println("Time: Load -> " + dataList!!.count())
    }

    fun resetList(){
        dataList!!.clear()
    }

    fun insertData() = runBlocking{
        val startTime = System.nanoTime();

        val job = launch {
            for (d in dataList!!) {
                database.dataDao().insert(d)
            }
        }
        job.join()

        val endTime = System.nanoTime();
        println("Time: Insert -> " + ((endTime - startTime) / 1000000))
        println("Time: Insert -> " + dataList!!.count())
    }

    fun deleteData() = runBlocking{
        val startTime = System.nanoTime();

        val job = launch {
            for (d in database.dataDao().all) {
                database.dataDao().delete(d)
            }
        }
        job.join()

        val endTime = System.nanoTime();
        println("Time: Delete -> " + ((endTime - startTime) / 1000000))
        println("Time: Delete -> " + database.dataDao().all.count())

        dataList!!.clear()
        addDataToList()

        println("Delete erfolgreich.")
    }

    fun addDataToList(){
        for(i in 1..1000)
            dataList!!.add(Data())
    }
}

File: bm_db_xam_platform/bm_db_xam_shared/MainActivityViewModel/MainActivityViewModel.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;
using System.Threading.Tasks;
using bm_db_xam_android.Helper;
using bm_db_xam_shared.Database;
using bm_db_xam_shared.Model;

namespace bm_db_xam_shared.ViewModel
{
    public class MainActivityViewModel
    {
        List<Data> DataList;
        static DataDatabase database;

        public MainActivityViewModel()
        {
            database = new DataDatabase(PathHelper.GetLocalPath("db_xam_platform"));
            DataList = new List<Data>();

            AddDataToList();
        }

        public async void LoadData()
        {
            var timer = new Stopwatch();
            timer.Start();

            DataList = await database.GetItemsAsync();

            timer.Stop();
            Debug.WriteLine("Time: Load -> " + timer.Elapsed);
            Debug.WriteLine("Time: Load -> " + DataList.Count);
        }

        public void ResetList()
        {
            DataList = null;
        }

        public void InsertData()
        {
            var timer = new Stopwatch();
            timer.Start();

            database.SaveItems(DataList);

            timer.Stop();
            Debug.WriteLine("Time: Insert -> " + timer.Elapsed);
            Debug.WriteLine("Time: Insert -> " + database.GetItemsAsync().Result.Count);
        }

        public void DeleteData()
        {
            var timer = new Stopwatch();
            timer.Start();

            database.DeleteItems(DataList);

            timer.Stop();
            Debug.WriteLine("Time: Insert -> " + timer.Elapsed);
            Debug.WriteLine("Time: Insert -> " + database.GetItemsAsync().Result.Count);

            DataList.Clear();
            AddDataToList();
        }

        private void AddDataToList()
        {
            for (int i = 0; i < 1000; i++)
            {
                DataList.Add(new Data());
            }
        }

    }
}
